include/revng/Pypeline/ObjectID.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/Pypeline/NanobindUtils.h"
+#include "revng/Pypeline/ObjectIDImpl.h"
+#include "revng/Pypeline/Registry.h"
+#include "revng/Pypeline/StringRefCaster.h"
+#include "revng/Pypeline/Utility.h"
+
+#include "nanobind/stl/optional.h"
```

Comment:
We put more "general" includes first.
Change `.clang-format` to put `nanobind/` includes after `boost` but before `llvm`.